### PR TITLE
US142491 Add siren Discussion Forum entity

### PR DESCRIPTION
**Pull request by mostlyfabulous**
[US142491](https://rally1.rallydev.com/#/?detail=/userstory/644998737805&fdp=true): [ui] Create choose forum dialog
[US141010](https://rally1.rallydev.com/#/?detail=/userstory/642314304651&fdp=true): [ui] Create new forum text

This let's us get the name of the topic's forum and add support for us to create a new forum as well.

### CODE REVIEW

**Comment by dustxd** [comment]
Can you add a test file for the forum entity as well?

**Review by dustxd — APPROVED** [review]

File: src/activities/discussions/DiscussionTopicEntity.js [hunk 1]
```diff
@@ -6,6 +6,17 @@ import { performSirenActions } from '../../es6/SirenAction.js';
  * DiscussionTopicEntity class representation of a D2L Discussion Topic.
  */
 export class DiscussionTopicEntity extends Entity {
+	/**
+	 * @returns {string} the href of the discussion topic's discussion forum entity
+	 */
+	forumHref() {
+		if (this._entity) {
+			const href = this._entity.getLinkByRel(Rels.Discussions.forum).href;
```
Comment (dustxd) [inline src/activities/discussions/DiscussionTopicEntity.js:14]: Might want to do a `this._entity.hasLinkByRel(Rels.Discussions.forum)` check here before getting the link

Comment (mostlyfabulous) [reply src/activities/discussions/DiscussionTopicEntity.js:14]: Ah yes, good shout

File: src/activities/discussions/DiscussionForumEntity.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	 * @param {object} forum the forum that's being modified
+	 * @returns {object} the appropriate action/fields to update
+	 */
+	_formatUpdateNameAction(forum) {
+		const { name } = forum || {};
+
+		if (!name) return;
+		if (!this._hasFieldValueChanged(name, this.name())) return;
+		if (!this.canEditName()) return;
+
```
[the remark this answers is not in the document]
Comment (dustxd) [reply src/activities/discussions/DiscussionForumEntity.js:34]: good catch

**Review by dustxd — APPROVED** [review]

**Pull request merged**